bundles/org.openhab.binding.hive/src/main/java/org/openhab/binding/hive/internal/client/adapter/Eui64GsonAdapter.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hive.internal.client.adapter;
+
+import java.io.IOException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.hive.internal.client.Eui64;
+
+import com.google.gson.stream.JsonReader;
+
+/**
+ * A gson {@link com.google.gson.TypeAdapter} for {@link Eui64}.
+ *
+ * @author Ross Brown - Initial contribution
+ */
+@NonNullByDefault
+public final class Eui64GsonAdapter extends SimpleGsonTypeAdapterBase<Eui64> {
+    @NonNullByDefault({})
```

Comment:
I've had a look and the reason I did this is because without it I get this error:
`Illegal redefinition of parameter in, inherited method from TypeAdapter<Eui64> does not constrain this parameter`
This means I would have to use `@Nullable JsonReader in` and check it.
I would never expect Gson to pass `null` to this method, and if I did I would just have to throw a `NullPointerException`/`IllegalArgumentException` anyway.